honeybee_energy/properties/model.py

```diff
@@ -261,6 +262,33 @@ def program_types(self):
                     program_types.append(room.properties.energy._program_type)
         return list(set(program_types))  # catch equivalent program types
 
+    def building_idf(self, solar_distribution='FullInteriorAndExteriorWithReflections'):
```

Comment:
What do you mean? Do you just mean that we should use a namedtuple to check that the input is valid (instead of staying `assert solar_distribution in ShadowCalculation.SOLAR_DISTRIBUTIONS`)?

Reply:
Not just for checking. I was thinking that we should create an object for `SolarDistribution` then use that for input. So instead of `FullInteriorAndExteriorWithReflections` you pass `SolarDistribution.FullInteriorAndExteriorWithReflections`.

Now that I think about it enumerate module is not supported in Python 2 so it will be a pain to handle it. Let's leave it as is.